src/js/libcs/OpSound.js

```diff
@@ -1,48 +1,533 @@
 //*******************************************************
 // and here are the definitions of the sound operators
 //*******************************************************
+var lines = [0];
+var linesDict = {0: 0};
+
+var getAudioContext = function() {
+    var ac = null;
+    if (!window.AudioContext && !window.webkitAudioContext) {
+        console.warn('Web Audio API not supported in this browser');
+    } else {
+        ac = new(window.AudioContext || window.webkitAudioContext)();
+    }
+    return function() {
+        return ac;
+    };
+}();
+
+var audioCtx = null;
 
-var sound = {};
-sound.lines = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+
+evaluator.stopsound$0 = function() {
+    audioCtx.close();
+    audioCtx = null;
+};
 
 evaluator.playsin$1 = function(args, modifs) {
+    if (!audioCtx) {
+        audioCtx = getAudioContext();
+    }
 
     function handleModifs() {
         var erg;
+        if (modifs.amp !== undefined) {
```

Comment:
maybe you can avoid some redundancy by using a helper function here. (`damp = realmodif("damp", 0 [default value])` etc.)